src/HackF5.UnitySpy/Detail/ProcessFacade.cs

```diff
@@ -189,26 +229,28 @@ private object[] ReadManagedArray(TypeInfo type, uint address)
             var arrayDefinition = type.Image.GetTypeDefinition(arrayDefinitionPtr);
             var elementDefinition = type.Image.GetTypeDefinition(this.ReadPtr(arrayDefinitionPtr));
 
-            var count = this.ReadInt32(ptr + 0xc);
-            var start = ptr + 0x10;
+            var elementSize = MonoLibraryOffsets.UsesArrayDefinitionSize ? arrayDefinition.Size : SizeOfPtr;
+
+            var count = this.ReadInt32(ptr + SizeOfPtr * 3);
+            var start = ptr + SizeOfPtr * 4;
```

Comment:
I would love to have a short explanation of the *3 and *4 here, if it's not too much of a bother to do